chain-signatures/node/src/protocol/triple.rs

```diff
@@ -170,82 +170,59 @@ impl TripleManager {
             .unwrap_or(false)
     }
 
-    async fn take(&mut self, id: &TripleId) -> Result<Triple, GenerationError> {
-        if self.contains_mine(id).await {
-            tracing::error!(?id, "cannot take mine triple as foreign owned");
-            return Err(GenerationError::TripleDenied(
-                *id,
-                "cannot take mine triple as foreign owned",
-            ));
-        }
-
-        let result = self.triple_storage.take(id).await.map_err(|store_err| {
-            if self.generators.contains_key(id) {
-                tracing::warn!(id, ?store_err, "triple is generating");
-                GenerationError::TripleIsGenerating(*id)
-            } else if self.gc.contains_key(id) {
-                tracing::warn!(id, ?store_err, "triple is garbage collected");
-                GenerationError::TripleIsGarbageCollected(*id)
-            } else {
-                tracing::warn!(id, ?store_err, "triple is missing");
-                GenerationError::TripleIsMissing(*id)
-            }
-        });
-
-        if result.is_ok() {
-            self.gc.insert(*id, Instant::now());
-        }
-
-        result
-    }
-
     /// Take two unspent triple by theirs id with no way to return it. Only takes
     /// if both of them are present.
     /// It is very important to NOT reuse the same triple twice for two different
     /// protocols.
     pub async fn take_two(
         &mut self,
-        id0: TripleId,
-        id1: TripleId,
+        id0: &TripleId,
+        id1: &TripleId,
```

Comment:
`TripleId` implements `Copy` so no need to pass by reference. It's also cheaper to pass it by value than reference here too since pointers are fat (i.e. 8 bytes in 64bit systems)